src/PostgREST/Error.hs

```diff
@@ -217,36 +218,17 @@ instance JSON.ToJSON SQL.QueryError where
 
 instance JSON.ToJSON SQL.CommandError where
   toJSON (SQL.ResultError (SQL.ServerError c m d h)) = JSON.object [
-        "code"    .= (T.decodeUtf8 c      :: Text),
-        "message" .= (T.decodeUtf8 m      :: Text),
-        "details" .= (fmap T.decodeUtf8 d :: Maybe Text),
-        "hint"    .= (fmap T.decodeUtf8 h :: Maybe Text)]
-
-  toJSON (SQL.ResultError (SQL.UnexpectedResult m)) = JSON.object [
-    "code"    .= HasqlErrorCode00,
-    "message" .= (m :: Text),
-    "details" .= JSON.Null,
-    "hint"    .= JSON.Null]
-  toJSON (SQL.ResultError (SQL.RowError i SQL.EndOfInput)) = JSON.object [
-    "code"    .= HasqlErrorCode01,
-    "message" .= ("Row error: end of input" :: Text),
-    "details" .= ("Attempt to parse more columns than there are in the result" :: Text),
-    "hint"    .= (("Row number " <> show i) :: Text)]
-  toJSON (SQL.ResultError (SQL.RowError i SQL.UnexpectedNull)) = JSON.object [
-    "code"    .= HasqlErrorCode02,
-    "message" .= ("Row error: unexpected null" :: Text),
-    "details" .= ("Attempt to parse a NULL as some value." :: Text),
-    "hint"    .= (("Row number " <> show i) :: Text)]
-  toJSON (SQL.ResultError (SQL.RowError i (SQL.ValueError d))) = JSON.object [
-    "code"    .= HasqlErrorCode03,
-    "message" .= ("Row error: Wrong value parser used" :: Text),
-    "details" .= d,
-    "hint"    .= (("Row number " <> show i) :: Text)]
-  toJSON (SQL.ResultError (SQL.UnexpectedAmountOfRows i)) = JSON.object [
-    "code"    .= HasqlErrorCode04,
-    "message" .= ("Unexpected amount of rows" :: Text),
-    "details" .= i,
+    "code"    .= (T.decodeUtf8 c      :: Text),
+    "message" .= (T.decodeUtf8 m      :: Text),
+    "details" .= (fmap T.decodeUtf8 d :: Maybe Text),
+    "hint"    .= (fmap T.decodeUtf8 h :: Maybe Text)]
+
+  toJSON (SQL.ResultError resultError) = JSON.object [
+    "code"    .= InternalErrorCode00,
+    "message" .= (show resultError :: Text),
+    "details" .= ("Internal error." :: Text),
```

Comment:
I think a `null` would be preferable to this since it doesn't really add more "details".